pkg/manifests/configfile.go

```diff
@@ -66,20 +86,119 @@ func NewConfigFile(path, workingDir string) (*ConfigFile, error) {
+	if cf.PatchUpdated != nil {
+		finalBytes, _, _, err := cf.getGeneratedAndPatchedManifests(ctx, manifests)
+		return finalBytes, err
+	}
+	return cf.getGeneratedManifests(ctx, manifests, cf.CommandUpdated.Generators)
+}
+
+func (cf *ConfigFile) SetWorkloadContainerImage(ctx context.Context, manifests Manifests, r resource.Resource, container string, newImageID image.Ref) error {
+	if cf.PatchUpdated != nil {
+		return cf.updatePatchFile(ctx, manifests, func(previousManifests []byte) ([]byte, error) {
+			return manifests.SetWorkloadContainerImage(previousManifests, r.ResourceID(), container, newImageID)
+		})
+	}
+
+	// Command-updated
+	result := cf.execContainerImageUpdaters(ctx, r.ResourceID(), container, newImageID.Name.String(), newImageID.Tag)
+	if len(result) == 0 {
+		return makeNoCommandsRunErr("update.containerImage", cf)
+	}
+
+	if len(result) > 0 && result[len(result)-1].Error != nil {
+		updaters := cf.CommandUpdated.Updaters
+		return fmt.Errorf("error executing image updater command %q from file %q: %s\noutput:\n%s",
+			updaters[len(result)-1].ContainerImage.Command,
+			result[len(result)-1].Error,
+			r.Source(),
+			result[len(result)-1].Output,
+		)
+	}
+	return nil
+}
+
+// UpdateWorkloadPolicies updates policies for a workload, using
+// commands or patching according to the config file.
+func (cf *ConfigFile) UpdateWorkloadPolicies(ctx context.Context, manifests Manifests, r resource.Resource, update resource.PolicyUpdate) (bool, error) {
+	if cf.PatchUpdated != nil {
+		var changed bool
+		err := cf.updatePatchFile(ctx, manifests, func(previousManifests []byte) ([]byte, error) {
+			updatedManifests, err := manifests.UpdateWorkloadPolicies(previousManifests, r.ResourceID(), update)
+			if err == nil {
+				changed = bytes.Compare(previousManifests, updatedManifests) != 0
+			}
+			return updatedManifests, err
+		})
+		return changed, err
+	}
+
+	// Command-updated
+	workload, ok := r.(resource.Workload)
+	if !ok {
+		return false, errors.New("resource " + r.ResourceID().String() + " does not have containers")
+	}
+	changes, err := resource.ChangesForPolicyUpdate(workload, update)
+	if err != nil {
+		return false, err
+	}
+
+	for key, value := range changes {
+		result := cf.execPolicyUpdaters(ctx, r.ResourceID(), key, value)
+		if len(result) == 0 {
+			return false, makeNoCommandsRunErr("updaters.policy", cf)
+		}
+
+		if len(result) > 0 && result[len(result)-1].Error != nil {
+			updaters := cf.CommandUpdated.Updaters
+			err := fmt.Errorf("error executing annotation updater command %q from file %q: %s\noutput:\n%s",
+				updaters[len(result)-1].Policy.Command,
+				result[len(result)-1].Error,
+				r.Source(),
+				result[len(result)-1].Output,
+			)
+			return false, err
+		}
+	}
+	// We assume that the update changed the resource. Alternatively, we could generate the resources
+	// again and compare the output, but that's expensive.
```

Comment:
Uhmm, we could simply use Git for this, I think. I wonder why I didn't think about it before (I know, not part of the PR)

Reply:
Right; I think this was probably ported over from the more circumspect implementation.